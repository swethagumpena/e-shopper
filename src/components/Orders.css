.total-data p{
    text-align: center;
    font-weight: bold;
    font-size: larger;
}