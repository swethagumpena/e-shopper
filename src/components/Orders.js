import React from 'react';
import PropTypes from 'prop-types';

const Orders = ({ orders }) => (
  <div>
    {/* <p>{JSON.stringify(orders)}</p> */}
    <div className="basket-msg">
      All Orders
      <hr />
      Past Orders (
      {orders.length}
      )
    </div>
    <div>
      <table>
        {orders.map((order) => (
          <>
            <tr key={order.id}>
              <th>ORDER</th>
              <th>ITEMS</th>
              <th>DATE</th>
              <th>AMOUNT</th>
            </tr>
            <tr>
              <td>
                Order id:
                {' '}
                {order.orderId}
              </td>
              <td>
                {order.itemsCount}
                {' '}
                items
              </td>
              <td>{order.date}</td>
              <td>{order.amount}</td>
            </tr>
            <tr>
              <th>ITEM DESCRIPTION</th>
              <th>UNIT PRICE</th>
              <th>QUANTITY</th>
              <th>SUB TOTAL</th>
            </tr>
            <tr className="category">
              <td>FRUITS</td>
              <td />
              <td />
              <td />
            </tr>
            {order.items.map((eachItem) => (
              <>
                <tr key={eachItem.id}>
                  <td>{eachItem.productName}</td>
                  <td className="items-align">
                    Rs.
                    {' '}
                    {eachItem.price}
                    .00
                  </td>
                  <td className="items-align">{eachItem.quantity}</td>
                  <td className="items-align">
                    Rs.
                    {' '}
                    {eachItem.price * eachItem.quantity}
                    .00
                  </td>
                </tr>
                <tr>
                  <td>
                    TOTAL Rs.
                    {' '}
                    {order.items.reduce((accumulator, item) => (
                      accumulator + eachItem.price * item.quantity), 0)}
                    .00
                  </td>
                </tr>

              </>
            ))}

          </>
        ))}
      </table>
    </div>

  </div>
);

const productShape = {
  id: PropTypes.number,
  productName: PropTypes.string,
  quantity: PropTypes.number,
  price: PropTypes.price,
  url: PropTypes.string,
};

Orders.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.shape(productShape)).isRequired,
};

export default Orders;
