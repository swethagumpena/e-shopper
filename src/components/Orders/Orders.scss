@import '../../styles/index.scss';

.total-data {
    p{
        text-align: center;
        font-weight: bold;
        font-size: larger;
    }
}
