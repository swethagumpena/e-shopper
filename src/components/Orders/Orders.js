import React from 'react';
import PropTypes from 'prop-types';
import './Orders.css';

const Orders = ({ orders }) => (
  <div>
    {/* <p>{JSON.stringify(orders)}</p> */}
    <div className="basket-msg">
      All Orders
      <hr />
      Past Orders (
      {orders.length}
      )
    </div>
    <div>
      <table className="basket-table">
        <tbody>
          {orders.map((order) => (
            <>
              <tr key={order.id}>
                <th>ORDER</th>
                <th>ITEMS</th>
                <th>DATE</th>
                <th>AMOUNT</th>
              </tr>
              <tr>
                <td className="items-align">
                  Order id:
                  {' '}
                  {order.orderId}
                </td>
                <td className="items-align">
                  {order.itemsCount}
                  {' '}
                  items
                </td>
                <td className="items-align">{order.date}</td>
                <td className="items-align">
                  Rs.
                  {' '}
                  {order.amount}
                  .00
                </td>
              </tr>
              <tr>
                <th>ITEM DESCRIPTION</th>
                <th>UNIT PRICE</th>
                <th>QUANTITY</th>
                <th>SUB TOTAL</th>
              </tr>
              <tr className="category">
                <td>FRUITS</td>
                <td />
                <td />
                <td />
              </tr>
              {order.items.map((eachItem) => (
                <>
                  <tr key={eachItem.id}>
                    <td>{eachItem.productName}</td>
                    <td className="items-align">
                      Rs.
                      {' '}
                      {eachItem.price}
                      .00
                    </td>
                    <td className="items-align">{eachItem.quantity}</td>
                    <td className="items-align">
                      Rs.
                      {' '}
                      {eachItem.price * eachItem.quantity}
                      .00
                    </td>
                  </tr>

                </>
              ))}
              <tr>
                <td />
                <td />
                <td />
                <td className="total-data">
                  <p>
                    TOTAL Rs.
                    {' '}
                    {order.items.reduce((accumulator, item) => (
                      accumulator + item.price * item.quantity), 0)}
                    .00
                  </p>
                </td>
              </tr>
            </>
          ))}
        </tbody>

      </table>
    </div>

  </div>
);

const productShape = {
  id: PropTypes.number,
  productName: PropTypes.string,
  quantity: PropTypes.number,
  price: PropTypes.number,
  url: PropTypes.string,
};

Orders.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.shape(productShape)).isRequired,
};

export default Orders;
