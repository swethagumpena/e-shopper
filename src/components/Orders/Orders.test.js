import {
  render, screen,
} from '@testing-library/react';
import React from 'react';
import Orders from './Orders';

describe(Orders, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const mockOrders = [{
    items: [
      {
        id: 7,
        name: 'banana',
        price: 5,
        count: 46,
        category: 'Fruits & Vegatables',
        inCartCount: 2,
      },
    ],
    id: 1,
    date: 1615525171927,
  }];

  test('should match snapshot', () => {
    const { container } = render(<Orders orders={mockOrders} />);
    expect(container).toMatchSnapshot();
  });

  test('should display All orders', () => {
    render(<Orders orders={mockOrders} />);
    screen.logTestingPlaygroundURL();
    expect(screen.getByText('All Orders'));
  });

  test('should display Past orders and number of orders', () => {
    render(<Orders orders={mockOrders} />);
    expect(screen.getByText('Past Orders (1)'));
  });

  test('should display all orders table', () => {
    render(<Orders orders={mockOrders} />);
    expect(screen.getByTestId('all-orders-table'));
  });
});
