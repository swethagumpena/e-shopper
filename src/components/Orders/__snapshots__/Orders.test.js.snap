// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match snapshot 1`] = `
<div>
  <div>
    <div
      class="basket-msg"
    >
      <h3>
        All Orders
      </h3>
      <hr />
      Past Orders (1)
    </div>
    <div>
      <table
        class="basket-table"
        data-testid="all-orders-table"
      >
        <tbody>
          <tr>
            <th>
              ORDER
            </th>
            <th>
              ITEMS
            </th>
            <th>
              DATE
            </th>
            <th>
              AMOUNT
            </th>
          </tr>
          <tr>
            <td
              class="items-align"
            >
              Order id: 1
            </td>
            <td
              class="items-align"
            >
              1 items
            </td>
            <td
              class="items-align"
            >
              12 - 2 - 2021
            </td>
            <td
              class="items-align"
            >
              Rs. 0.00
            </td>
          </tr>
          <tr>
            <th>
              ITEM DESCRIPTION
            </th>
            <th>
              UNIT PRICE
            </th>
            <th>
              QUANTITY
            </th>
            <th>
              SUB TOTAL
            </th>
          </tr>
          <tr>
            <td>
              banana
            </td>
            <td
              class="items-align"
            >
              Rs. 5.00
            </td>
            <td
              class="items-align"
            >
              2
            </td>
            <td
              class="items-align"
            >
              Rs. 10.00
            </td>
          </tr>
          <tr>
            <td />
            <td />
            <td />
            <td
              class="total-data"
            >
              <p>
                TOTAL Rs. 10.00
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
